pkgs/servers/gonic/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib, buildGoPackage, fetchFromGitHub
+, pkg-config, taglib, alsaLib
+
+# Disable on-the-fly transcoding,
+# removing the dependency on ffmpeg.
+# The server will (as of 0.11.0) gracefully fall back
+# to the original file, but if transcoding is configured
+# that takes a while. So best to disable all transcoding
+# in the configuration if you disable transcodingSupport.
+, transcodingSupport ? true, ffmpeg
+
+# udpater
+, writers, vgo2nix }:
+
+assert transcodingSupport -> ffmpeg != null;
+
+let
+  # update these, then run `updateScript` to update dependencies
+  version = "0.11.0";
+  rev = "056fb54a703ef5b5194ce112cbbdd8fb53dbb1ea";
+  sha256 = "0hd794wrz29nh89lfnq67w1rc23sg085rqf1agwlgpqycns2djl9";
+
+  src = fetchFromGitHub {
+    owner = "sentriz";
+    repo = "gonic";
+    # 0.10.0
+    inherit rev sha256;
+  };
+
+in
+buildGoPackage {
+  pname = "gonic-${version}";
+  inherit version src;
+  goPackagePath = "go.senan.xyz/gonic";
+  goDeps = ./deps.nix;
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ taglib alsaLib ];
+
+  patchPhase = lib.optionalString transcodingSupport ''
+    substituteInPlace \
+       server/encode/encode.go \
+      --replace \
+        'ffmpegPath = "/usr/bin/ffmpeg"' \
+        'ffmpegPath = "${ffmpeg}/bin/ffmpeg"' \
+  '';
+
+  passthru.updateScript = writers.writeDash "update-gonic" ''
+    ${vgo2nix}/bin/vgo2nix \
```

Comment:
Ideally, `updateScript` should also find the latest version and update the hash in the source file.

`genericUpdater` might help https://github.com/NixOS/nixpkgs/pull/85304#issuecomment-614121523

Reply:
Is this documented somewhere? I can’t find anything. It’s only used in a few places.

Reply:
Not yet, outside of the comment I linked.

You would use something like:

```
  passthru.updateScript = writers.writeDash "update-gonic" ''
    ${pkgs.genericUpdater {
      inherit pname version;
      versionLister = "${pkgs.common-updater-scripts}/bin/list-git-tags ${lib.escapeShellArg sr.meta.homepage}";
    }}
    ${vgo2nix}/bin/vgo2nix …
  '';
```

But that would not be enough if the `rev` needs to be updated too.

Reply:
Cool, once it’s documented I will take a look.